src/inmanta/resources.py

```diff
@@ -577,13 +576,21 @@ def __init__(self, _id: "Id") -> None:
         self._references: dict[uuid.UUID, references.Reference[references.RefValue]] = {}
         self._resolved = False
 
+    def get(self, key: str, default: object = None) -> object:
+        if key in self.fields:
+            return getattr(self, key)
+        return default
```

Comment:
This is how you make it work? Can't we convert to a dict and back? Or at least implement the full `MutableMapping` protocol instead of `get` and `__contains__`?
That might also get rid of the mypy error. I tried to verify but I get `Skipping analyzing "jsonpath_ng": module is installed, but missing library stubs or py.typed marker`. Did you install a custom types package that you forgot to add to dev requires?